astroquery/eso/core.py

```diff
@@ -695,16 +729,63 @@ def retrieve_data(self, datasets, continuation=False, destination=None):
                     raise RemoteServiceError("There was a remote service "
                                              "error; perhaps the requested "
                                              "file could not be found?")
-            for fileId in root.select('input[name=fileId]'):
-                log.info("Downloading file {0}...".format(fileId.attrs['value'].split()[0]))
-                fileLink = ("http://dataportal.eso.org/dataPortal" +
-                            fileId.attrs['value'].split()[1])
+
+            if with_calib != 'none':
+                # when requested files with calibrations, some javascript is
+                # used to display the files, which prevent retrieving the files
+                # directly. So instead we retrieve the download script provided
+                # in the web page, and use it to extract the list of files.
+                # The benefit of this is also that in the download script the
+                # list of files is de-duplicated, whereas on the web page the
+                # calibration files would be duplicated for each exposure.
+                try:
+                    link = root.select('a[href$=/script]')[0]
+                    if 'downloadRequest' not in link.text:
+                        # Make sure that we found the correct link
+                        raise Exception
+                    href = link.attrs['href']
+                    script = self._request("GET", href, cache=False)
+                    fileLinks = re.findall(
+                        r'"(https://dataportal.eso.org/dataPortal/api/requests/.*)"',
+                        script.content.decode('utf8'))
+
+                    # links with api/ do not work, wtf ???
```

Comment:
...really?!  wtf indeed.  Maybe report this upstream?

Reply:
Yes, that's weird, without replacing I get an error:

> HTTPError: 500 Server Error: Internal Server Error for url: https://dataportal.eso.org/dataPortal/api/requests/...

There must be something done differently in the download script, but it is not obvious what. 
Do you know where to report this?

Reply:
Maybe just the ESO helpdesk?  They'll direct you to the right person

Reply:
Actually, thinking more about this, this is just because the authentication is managed differently with the two urls: with `https://dataportal.eso.org/dataPortal/api` you need to use Basic Authentication (which the script does), though with `https://dataportal.eso.org/dataPortal/` you can reuse the request session with the cookie based (?) authentication.
So I can make it work with the `api/` urls if I use something like this:
 ```
+            from requests.auth import HTTPBasicAuth
+            auth = HTTPBasicAuth(self.USERNAME,
+                                 keyring.get_password("astroquery:www.eso.org",
+                                                      self.USERNAME))
+
             log.debug("Files:\n{}".format('\n'.join(fileLinks)))
             for fileLink in fileLinks:
                 fileId = fileLink.rsplit('/', maxsplit=1)[1]
                 log.info("Downloading file {0}...".format(fileId))
                 filename = self._request("GET", fileLink, save=True,
-                                         continuation=True)
+                                         continuation=True, auth=auth)
```
But it's better to modify the urls and reuse the request session (the above supposes that the username/password is stored).